File: src/test/kotlin/com/example/mortgage/repository/CustomerRepositoryTest.java
package com.example.mortgage.repository;

import com.example.mortgage.model.Customer;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.test.context.junit4.SpringRunner;

import java.util.List;
import java.util.Optional;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

@RunWith(SpringRunner.class)
@DataJpaTest
public class CustomerRepositoryTest {

    @Autowired
    private CustomerRepository repository;

    private final long id = 2L;

    @Test
    public void testFindAll() {

        List<Customer> customers = repository.findAll();

        assertNotNull(customers);

        for (Customer customer : customers) {
            System.out.println("customer = " + customer);
        }
    }

    @Test
    public void testFindById() {

        Optional<Customer> entity = repository.findById(id);

        assertTrue(entity.isPresent());

        Customer customer = entity.get();

        assertEquals(id, customer.getCustomerId());
        assertEquals("Boba", customer.getFirstName());
        assertEquals("Loo", customer.getLastName());
        assertEquals("8015556874", customer.getPhone());
        assertEquals("bobaloo@live.com", customer.getEmail());
    }

    @Test
    public void testExistsById() {

        assertTrue(repository.existsById(id));
        assertFalse(repository.existsById(1234L));
    }

    @Test
    public void testCount() {

        assertEquals(4, repository.count());
    }

}

File: src/test/kotlin/com/example/mortgage/repository/CustomerRepositoryTest.kt
package com.example.mortgage.repository

import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
import org.springframework.test.context.junit4.SpringRunner

@RunWith(SpringRunner::class)
@DataJpaTest
class CustomerRepositoryTest {
    @Autowired
    private val repository: CustomerRepository? = null
    private val id = 2L

    @Test
    fun testFindAll() {
        val customers = repository!!.findAll()
        Assert.assertNotNull(customers)
        for (customer in customers) {
            println("customer = $customer")
        }
    }

    @Test
    fun testFindById() {
        val entity = repository!!.findById(id)
        Assert.assertTrue(entity.isPresent)
        val (customerId, firstName, lastName, phone, email) = entity.get()
        Assert.assertEquals(id, customerId)
        Assert.assertEquals("Boba", firstName)
        Assert.assertEquals("Loo", lastName)
        Assert.assertEquals("8015556874", phone)
        Assert.assertEquals("bobaloo@live.com", email)
    }

    @Test
    fun testExistsById() {
        Assert.assertTrue(repository!!.existsById(id))
        Assert.assertFalse(repository.existsById(1234L))
    }

    @Test
    fun testCount() {
        Assert.assertEquals(4, repository!!.count())
    }
}
